verify/common.go

```diff
@@ -35,146 +35,351 @@ type ErrWithHelp interface {
 	Help() string
 }
 
+type logger struct{}
+
+func (logger) errorf(fmtStr string, args ...interface{}) {
+	fmt.Printf("::error::"+fmtStr+"\n", args...)
+}
+
+func (logger) debugf(fmtStr string, args ...interface{}) {
+	fmt.Printf("::debug::"+fmtStr+"\n", args...)
+}
+
+func (logger) warningf(fmtStr string, args ...interface{}) {
+	fmt.Printf("::warning::"+fmtStr+"\n", args...)
+}
+
+var log logger
+
 type PRPlugin struct {
-	ForAction func(string) bool
 	ProcessPR func(pr *github.PullRequest) (string, error)
-	Name string
-	Title string
+	Name      string
+	Title     string
+
+	logger
 }
 
-func (p *PRPlugin) Entrypoint(env *ActionsEnv) error {
-	if p.ForAction != nil && !p.ForAction(env.Event.GetAction()) {
-		return nil
+func (p PRPlugin) processPR(pr *github.PullRequest) (conclusion, summary, text string, err error) {
+	text, err = p.ProcessPR(pr)
+
+	if err != nil {
+		conclusion = "failure"
+		summary = err.Error()
+		var helpErr ErrWithHelp
+		if errors.As(err, &helpErr) {
+			text = helpErr.Help()
+		}
+	} else {
+		conclusion = "success"
+		summary = "Success"
 	}
 
-	repoParts := strings.Split(env.Event.GetRepo().GetFullName(), "/")
-	orgName, repoName := repoParts[0], repoParts[1]
-	
-	headSHA := env.Event.GetPullRequest().GetHead().GetSHA()
-	fmt.Printf("::debug::creating check run %q on %s/%s @ %s...\n", p.Name, orgName, repoName, headSHA)
+	// Log in case we can't submit the result for some reason
+	p.debugf("plugin conclusion: %q", conclusion)
+	p.debugf("plugin result summary: %q", summary)
+	p.debugf("plugin result details: %q", text)
 
-	resRun, runResp, err := env.Client.Checks.CreateCheckRun(context.TODO(), orgName, repoName, github.CreateCheckRunOptions{
-		Name: p.Name,
-		HeadSHA: headSHA,
-		Status: github.String("in_progress"),
-	})
+	return
```

Comment:
please explicitly return things -- bare returns are a great way to cause issues for future maintainers, IMO

```suggestion
	return conclusion, summary, text, err
```